web/src/AppController.ts

```diff
@@ -116,6 +120,28 @@ class AppController {
       this.createNewSocket()
     }, timeout)
   }
+
+  setStateFromSnapshot(): void {
+    let url = this.url
+    fetch(url)
+      .then(resp =>
+        resp
```

Comment:
nested promise chains like this are strongly discouraged. a good rule of them is that whenever you have a `then` call inside a `then` handler, you're probably doing something wrong

try something like:
```
fetch(url).then(resp => resp.json()).then(data => ...)
```